### PR TITLE
fix: deeplink

### PR DESCRIPTION
**Pull request by guipasmoi**
here shot at fixing deep linking
You need check for IOS it may be not working
PS there is an error on the map (present on master and here too)

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
           <action android:name="android.intent.action.VIEW"/>
           <category android:name="android.intent.category.DEFAULT"/>
           <category android:name="android.intent.category.BROWSABLE"/>
-          <data android:scheme="strava" android:host="localhost" />
```
Comment (guipasmoi) [inline android/app/src/main/AndroidManifest.xml:34]: from env file

File: app/hocs/deeplink.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    componentDidMount() {
+      Linking.getInitialURL().then(this.handleOpenURL).catch(err => {
+        // eslint no-console: "error"
+        console.error("[Deep linking] An error occurred ", err);
+      });
+      Linking.addEventListener("url", this.handleOpenURL);
+    }
+
+    componentWillUnmount() {
+      Linking.removeEventListener("url", this.handleOpenURL);
```
Comment (guipasmoi) [inline app/hocs/deeplink.js:24]: remove correctly now thx to hoc

File: app/hocs/deeplink.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    handleOpenURL = event => {
+      if (event && event.url) {
+        const navigationAction = WrappedComponent.router.getActionForPathAndParams(
```
Comment (guipasmoi) [inline app/hocs/deeplink.js:29]: here the magic happen to navigate to any route

File: app/hocs/deeplink.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          getPathAndQuery(event.url)
+        );
+        if (navigationAction) {
+          this.context.store.dispatch(navigationAction);
+        } else {
+          console.warn(
+            `[Deep linking] couldn't find a matching route for the url sheme "${event.url}"`
```
Comment (guipasmoi) [inline app/hocs/deeplink.js:36]: 💃

File: app/hocs/deeplink.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        }
+      }
+    };
+
+    render() {
+      return <WrappedComponent {...this.props} />;
+    }
+  }
+
+  return hoistStatics(DeeplinkingContainer, WrappedComponent);
```
Comment (guipasmoi) [inline app/hocs/deeplink.js:47]: expose statics of the wrappedcomponent (react-redux use it)

File: app/routes/AppNavigator.js [hunk 6]
```diff
@@ -1,32 +1,43 @@
...
+};
 
 const AppNavigator = StackNavigator(
   {
-    Login: { screen: Login },
-    Authenticate: { screen: Authenticate, path: "localhost" },
-    Home: { screen: Home }
+    Login: {
+      screen: Login,
+      path: "login"
```
Comment (guipasmoi) [inline app/routes/AppNavigator.js:21]: define the path for login

File: app/routes/AppNavigator.js [hunk 7]
```diff
@@ -1,32 +1,43 @@
...
+    },
+    Main: {
+      screen: Main,
+      path: "main"
```
Comment (guipasmoi) [inline app/routes/AppNavigator.js:25]: same

File: app/routes/screens/login/Login.js [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
-
 import { View, WebView } from "react-native";
-
 import Config from "react-native-config";
 
 import styles from "./styles";
 
-const INITIAL_URI = `https://www.strava.com/oauth/authorize?client_id=${Config.CLIENT_ID}&response_type=code&redirect_uri=strava://localhost&scope=public`;
+const { CLIENT_ID, URL_SHEME_PREFIX, URL_SHEME_HOST } = Config;
+const CALL_BACK_URL_SHEME = `${URL_SHEME_PREFIX}://${URL_SHEME_HOST}/main`;
```
Comment (guipasmoi) [inline app/routes/screens/login/Login.js:8]: 😍  no strange constants

File: app/routes/screens/login/Login.js [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
+const INITIAL_URI = `https://www.strava.com/oauth/authorize?client_id=${CLIENT_ID}&response_type=code&redirect_uri=${CALL_BACK_URL_SHEME}&scope=public`;
```
Comment (guipasmoi) [inline app/routes/screens/login/Login.js:9]: should extract that one with the api constants 
may be we should create a wrapper arround config to expose computed configs

File: app/routes/screens/main/Main.js [hunk 10]
```diff
@@ -3,12 +3,12 @@ import React from "react";
...
-import ClubFeed from "../clubFeed/ClubFeed";
-import PerformanceMeter from "../performanceMeter/PerformanceMeter";
-import RacePredictor from "../race/Race";
+import Details from "./details/Details";
+import ClubFeed from "./clubFeed/ClubFeed";
+import PerformanceMeter from "./performanceMeter/PerformanceMeter";
+import RacePredictor from "./race/Race";
 
-const Home = TabNavigator(
+const Main = TabNavigator(
```
Comment (guipasmoi) [inline app/routes/screens/main/Main.js:11]: rename home in Main because its not home

**Comment by totorototo** [comment]
Map error is fixed on geolocation branch

**Comment by guipasmoi** [comment]
README .env file  isn't clear (->multiple part)
I THINK all keys in the .env are currently mandatory .
we should make it so if a key is not provided the features depending on it should be desactivated

**Comment by totorototo** [comment]
.env part has to be re-written actually. 
You're right, each part of it is now mandatory (strava, firebase, google maps, ....).
May be in a near future we could make feature depending on key definition. (you're right too)

File: app/routes/screens/main/Main.js (hunk 10 above)
Comment (totorototo) [reply app/routes/screens/main/Main.js:11]: 👍 makes sense

File: app/hocs/deeplink.js (hunk 2 above)
Comment (totorototo) [reply app/hocs/deeplink.js:24]: amazing! 🥇

File: app/hocs/deeplink.js (hunk 3 above)
Comment (totorototo) [reply app/hocs/deeplink.js:29]: 🎉

File: app/hocs/deeplink.js (hunk 4 above)
Comment (totorototo) [reply app/hocs/deeplink.js:36]: olé

File: app/hocs/deeplink.js (hunk 5 above)
Comment (totorototo) [reply app/hocs/deeplink.js:47]: dunno that!

File: app/routes/screens/login/Login.js (hunk 8 above)
Comment (totorototo) [reply app/routes/screens/login/Login.js:8]: prettier I would say 🤙

File: app/routes/screens/login/Login.js (hunk 9 above)
Comment (totorototo) [reply app/routes/screens/login/Login.js:9]: Had to think about that

**Comment by totorototo** [comment]
check doen on iOS. it seems to be ok!
Maps error is already fixed on sibling branch.
Ready to merge

**Review by totorototo — APPROVED** [review]
👍 I definitely love the hoc approach!

File: app/hocs/deeplink.js (hunk 5 above)
Comment (guipasmoi) [reply app/hocs/deeplink.js:47]: navigators -> need WrappedComponent statics thats how they works

**Pull request merged**